secant/Screens/Onboarding/OnboardingContentView.swift

```diff
@@ -61,34 +61,76 @@ struct OnboardingContentView: View {
                     VStack(spacing: viewStore.isFinalStep ? 50 : 15) {
                         HStack {
                             Text(viewStore.steps[stepIndex].title)
-                                .font(.custom(FontFamily.Roboto.bold.name, size: 30))
-                                .fontWeight(.regular)
+                                .titleText()
+                                .lineLimit(0)
+                                .minimumScaleFactor(0.1)
                             if !viewStore.isFinalStep {
                                 Spacer()
                             }
                         }
                         
                         Text(viewStore.steps[stepIndex].description)
-                            .font(.custom(FontFamily.Roboto.regular.name, size: 15))
-                            .lineSpacing(5)
+                            .synopsisText()
+                            .lineSpacing(2)
+                            .opacity(0.53)
                     }
                     .opacity(stepIndex == viewStore.index ? 1: 0)
                     .padding(.horizontal, 35)
                     .frame(width: width, height: height)
                 }
             }
-            .offset(y: viewStore.isFinalStep ? width / 2.3 : viewStore.offset + width / 2.3)
+            .offset(y: viewStore.isFinalStep ? width / 2.5 : viewStore.offset + height / descriptionOffsetCoefficient)
         }
     }
 }
 
+/// Following computations are necessary to handle properly sizing and positioning of elements
+/// on different devices (apects). iPhone SE and iPhone 8 are similar aspect family devices
+/// while iPhone X, 11, etc are different family devices, capable to use more of the space.
+extension OnboardingContentView {
+    var circularFrameUniformSize: CGFloat {
+        let aspect = height / width
+        let deviceMultiplier = 1.0 + (((aspect / 1.725) - 1.0) * 2.0)
+        
+        return width * (0.6 * deviceMultiplier)
+    }
+
+    var circularFrameOffsetCoeffcient: CGFloat {
+        let aspect = height / width
```

Comment:
will this crash if for some reason width ends up being Zero?

Reply:
it would, fixed